extensions/amp-social-share/0.2/amp-social-share.js

```diff
@@ -0,0 +1,176 @@
+        !!this.element.ownerDocument.querySelector(
+          'amp-social-share[type=system][data-mode=replace]'
+        );
+      if (systemOnly) {
+        return;
+      }
+    }
+    const typeConfig = getSocialConfig(type) || dict();
+    if (typeConfig['obsolete']) {
+      user().warn(TAG, `Skipping obsolete share button ${type}`);
+      return;
+    }
+    return typeConfig;
+  }
+
+  /**
+   * @private
+   * @param {!../../../src/service/viewer-interface.ViewerInterface} viewer
+   * @param {!../../../src/service/platform-impl.Platform} platform
+   * @return {boolean}
+   */
+  systemShareSupported_(viewer, platform) {
+    // Chrome exports navigator.share in WebView but does not implement it.
+    // See https://bugs.chromium.org/p/chromium/issues/detail?id=765923
+    const isChromeWebview = viewer.isWebviewEmbedded() && platform.isChrome();
+
+    return 'share' in navigator && !isChromeWebview;
+  }
+
+  /**
+   * Sets 'href' and 'target' in the AmpContext based on AMP-specific conditions.
+   * These needs to be passed to the context as opposed to as properties in `init()`
+   * because their values are resolved asynchronously.
+   *
+   * @private
+   * @param {!JsonObject} typeConfig
+   * @param {!../../../src/service/platform-impl.Platform} platform
+   */
+  setHrefAndTargetContext_(typeConfig, platform) {
+    const shareEndpoint = user().assertString(
+      this.element.getAttribute('data-share-endpoint') ||
+        typeConfig['shareEndpoint'],
+      'The data-share-endpoint attribute is required. %s'
+    );
+    const urlParams = getDataParamsFromAttributes(this.element);
+    Object.assign(urlParams, typeConfig['defaultParams']);
+    const hrefWithVars = addParamsToUrl(shareEndpoint, urlParams);
+    const urlReplacements = Services.urlReplacementsForDoc(this.element);
+    const bindingVars = typeConfig['bindings'];
+    const bindings = {};
+    if (bindingVars) {
+      bindingVars.forEach(name => {
+        const bindingName = name.toUpperCase();
+        bindings[bindingName] = urlParams[name];
+      });
+    }
+
+    urlReplacements.expandUrlAsync(hrefWithVars, bindings).then(result => {
+      let href = result;
+      // mailto:, sms: protocols breaks when opened in _blank on iOS Safari
+      const {protocol} = Services.urlForDoc(this.element).parse(href);
+      const isMailTo = protocol === 'mailto:';
+      const isSms = protocol === 'sms:';
+      this.context_['target'] =
+        platform.isIos() && (isMailTo || isSms)
+          ? '_top'
+          : this.element.getAttribute('data-target') || '_blank';
+      if (isSms) {
+        // http://stackoverflow.com/a/19126326
+        // This code path seems to be stable for both iOS and Android.
+        href = href.replace('?', '?&');
+      }
+      this.context_['href'] = href;
+    });
+  }
+}
+
+/** @override */
+AmpSocialShare['Component'] = SocialShare;
+
+/** @override */
+AmpSocialShare['props'] = {
+  'layout': {attr: 'layout'},
```

Comment:
Should `layout` be here? So far this was handled by AMP itself. It looks like `width` and `height` are still used in the component itself and I'll separate the discussion there.

Reply:
Removed `layout`